pkg/apk/apk/implementation.go

```diff
@@ -700,13 +707,13 @@ func (a *APK) InstallPackages(ctx context.Context, sourceDateEpoch *time.Time, a
 		g.Go(func() error {
 			exp, err := a.expandPackage(ctx, pkg)
 			if err != nil {
-				return fmt.Errorf("expanding %s: %w", pkg, err)
+				err = fmt.Errorf("expanding %s: %w", pkg, err)
```

Comment:
I would do something like:

```go
g.Go(func() (rerr error) {
  defer func() { done[i] <- rerr }()

  exp, err := a.expandPackage(ctx, pkg)
  if err != nil {
    return fmt.Errorf("expanding %s: %w", pkg, err)
  }

  expanded[i] = exp

  return nil
})
```

That creates a named return (yucky but useful for this error handling stuff) and closes `done[i]` with the returned error (which is nil on success).

Reply:
I might not love this approach actually because now we need buffered channels instead of unbuffered if we're going to send 🤔